data_structures/binary_tree/binary_search_tree.py

```diff
@@ -2,202 +2,177 @@
 A binary search Tree
 """
 
-
 class Node:
-    def __init__(self, label, parent):
-        self.label = label
+    def __init__(self, data, parent):
+        self.data = data
         self.left = None
         self.right = None
-        # Added in order to delete a node easier
-        self.parent = parent
+        self.parent = parent    # Added in order to delete a node easier
 
-    def getLabel(self):
-        return self.label
+    def getdata(self):
```

Comment:
Please use __get_data()__ instead of __getdata()__ for readability...  Same for other functions below.